exatomic/va/va.py

```diff
@@ -0,0 +1,327 @@
+                xyz['y'] *= Length['au', 'Angstrom']
+                xyz['z'] *= Length['au', 'Angstrom']
+                xyz.to_csv(p, header=False, index=False, sep=' ', float_format=float_format,
+                            quoting=csv.QUOTE_NONE, escapechar=' ')
+                p.write('\n')
+
+    def gen_slurm_inputs(self, path, sbatch, module, end_com=''):
+        """
+        Method to write slurm scripts to execute gradients and property calculations given
+        the displaced coordinates.
+
+        Method generates separate directories containing the slurm script for each calculation.
+        Will need to submit with some external shell script.
+
+        Need to define the module and sbatch variables to what is needed by each user. It was
+        built like this to make it the most general and applicable to more than one type of
+        quantum chemistry code.
+
+        Args:
+            path (str): path to where the directories will be generated and the inputs will
+                        will be read from
+            sbatch (dict): sbatch commands that are to be used for batch script execution
+            module (str): multiline string that will contain the module loading and other
+                          user specific variables
+            end_com (str): commands to be placed at the end of the slurm script
+        """
+        _name = "{job}{int}.dir"
+        _sbatch = "#SBATCH --{key}={value}"
+        files = os.listdir(path)
+        for file in files:
+            if file.endswith(".inp") and file.startswith("confo"):
+                fdx = file.replace("confo", "").replace(".inp", "")
+                job = "jobo"
+            elif file.endswith(".inp") and file.startswith("confg"):
+                fdx = file.replace("confg", "").replace(".inp","")
+                job = "jobg"
+            else:
+                continue
+            try:
+                os.mkdir(path+_name.format(job=job, int=fdx))
+                j_path = path+_name.format(job=job, int=fdx)
+            except OSError:
+                raise OSError("Failed to create directory {}".format(path+_name.format(
+                                                                                job=job, int=fdx)))
+            slurm = file.replace(".inp", ".slurm")
+            with open(path+file, 'r') as f:
+                with open(mkp(j_path, slurm), 'w') as j:
+                    j.write("#!/bin/bash\n")
+                    for key in sbatch.keys():
+                        j.write(_sbatch.format(key=key, value=sbatch[key])+'\n')
+                    j.write(_sbatch.format(key="job-name", value=file.replace(".inp", "")))
+                    j.write(_sbatch.format(key="output", value=file.replace(".inp", ".out")))
+                    j.write(module)
+                    for line in f:
+                        j.write(line)
+                    j.write(end_com+'\n')
+                    f.close()
+                    j.close()
```

Comment:
When using context managers, you don't need to explicitly call `close` - it is done for you.

Reply:
I felt like that was the case, but I was just trying to be safe with it.

Reply:
I have taken out the file.close commands with this last push.